### PR TITLE
Efgs documentation

### PR DESCRIPTION
**Pull request by solita-alperttiti**

### CODE REVIEW

File: exposure-notification/README.md [hunk 1]
```diff
@@ -92,6 +92,50 @@ Configurable values use [Duration parsing rules](https://docs.oracle.com/javase/
 1. Data access cache in RAM: Some status requests will always get through the proxy caches, so the most common database queries are cached in service RAM to ensure rapid responses.
     * covid19.diagnosis.data-cache.status-duration: database fetches needed for configuration or status information
 
+## European Federation Gateway Service (EFGS) integration
+
+EFGS is an official European solution for interoperability between European national COVID19 application backend servers.
+EFGS allows national backends change keys, which effectively means that national COVID-19 mobile tracing apps are compatible with each other.
```
Comment (jyrkija) [inline exposure-notification/README.md]: ... backends **to** change keys...

File: exposure-notification/README.md [hunk 2]
```diff
@@ -92,6 +92,50 @@ Configurable values use [Duration parsing rules](https://docs.oracle.com/javase/
...
+
+### Interchange of keys
+
+Same database table will be used for keys from other national backends than which is used for local keys. Consent of
+user willingness to share keys with other national backends will be stored on each indiviudal key. Same applies to
+user provided data of visited countries.  
+
+![efgs_inbound.png](../documentation/generated_images/efgs_inbound.png)
+
+* Inbound keys will be verified and validated on receive and then mixed with local keys in the database
```
Comment (jyrkija) [inline exposure-notification/README.md]: ...on **reception** and...

File: exposure-notification/README.md [hunk 3]
```diff
@@ -92,6 +92,50 @@ Configurable values use [Duration parsing rules](https://docs.oracle.com/javase/
...
+  * Inbound will be made based on callback request when it arrives or once in a day after UTC midnight
```
Comment (jyrkija) [inline exposure-notification/README.md]: Inbound **operation**...

File: exposure-notification/README.md [hunk 4]
```diff
@@ -92,6 +92,50 @@ Configurable values use [Duration parsing rules](https://docs.oracle.com/javase/
...
+  * Keys can be separated on database level by origin field. Origin will not be send to the mobile app, which sees
```
Comment (jyrkija) [inline exposure-notification/README.md]: ... **the** database level ... will not be sen**t** ...
Also, a long sentence with a double ", which" -> full stop somewhere would be prettier :)

File: documentation/plantuml/efgs_inbound.pu [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    VALIDATE --> STORE
+    STORE : Store keys to db
+    STORE --> FINISHED
+    STARTED --> ERROR
+    VERIFY --> FAIL
+    VALIDATE --> FAIL
+    ERROR --> FINISHED
+    ERROR --> ERROR
+    ERROR : Until max retries reached
+    FINISHED : Keys counts stored in db
```
Comment (jyrkija) [inline documentation/plantuml/efgs_inbound.pu]: Key counts...

File: documentation/plantuml/efgs_inbound.pu [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    VALIDATE : Validate batch data
+    VALIDATE --> STORE
+    STORE : Store keys to db
+    STORE --> FINISHED
+    STARTED --> ERROR
+    VERIFY --> FAIL
+    VALIDATE --> FAIL
+    ERROR --> FINISHED
+    ERROR --> ERROR
+    ERROR : Until max retries reached
```
Comment (jyrkija) [inline documentation/plantuml/efgs_inbound.pu]: The keyword is missing. Would "Retry until max attempts reached" be too long? Same in the other pic.

File: documentation/plantuml/efgs_outbound.pu [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    RECEIVED : Stored in db
+    RECEIVED : No operation created yet
+    RECEIVED --> STARTED
+    STARTED --> FINISHED
+    STARTED --> ERROR
+    STARTED : Efgs sync timestamp updated in db
+    ERROR --> FINISHED
+    ERROR --> ERROR
+    ERROR : Until max retries reached
+    FINISHED : Number of keys send stored to db
```
Comment (jyrkija) [inline documentation/plantuml/efgs_outbound.pu]: sen**t**

File: documentation/plantuml/efgs_inbound.pu [hunk 8]
```diff
@@ -0,0 +1,25 @@
+@startuml
```
Comment (jyrkija) [inline documentation/plantuml/efgs_inbound.pu:1]: I like this text-based picture thing. Have to remember these in other projects.

**Review by jyrkija — APPROVED** [review]

File: documentation/plantuml/efgs_inbound.pu (hunk 6 above)
Comment (solita-alperttiti) [reply documentation/plantuml/efgs_inbound.pu]: I think it's not. Changed that.

**Pull request merged**